profiles/auxiliaries/loki.nix

```diff
@@ -1,8 +1,12 @@
-_: {
+{ pkgs, ... }: {
   networking.firewall.allowedTCPPorts = [
     3100  # loki
   ];
 
+  # Required for Loki >= 2.3.0 ruler
+  systemd.services.loki.serviceConfig.ExecStartPre =
+    "+${pkgs.coreutils}/bin/mkdir -p /etc/loki/rules/fake";
```

Comment:
That could probably be handled with `systemd.tmpfiles.rules`

Reply:
From [man](https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html): "tmpfiles.d should be used for files whose lifetime is independent of any service".
By that defn it would seem preferable to leave this under the service, no?
Although maybe you're wondering about a corresponding execStartPost?

Reply:
Actually, I'm able to drop this stanza altogether now and re-direct the rules file to `/var/lib/loki/{rules,rules-tmp}` as part of the module extraConfig.  I think in the older versions I initially wrote this in, that wasn't working although I don't recall why.

This also brings up one other thought I'll follow up with you on.

Reply:
Sounds good. The reasoning behind using tmpfiles was that while it's independent of the service, in Nix we can add/remove the service and tmpfiles entry at the same time with a simple mkIf.
But your approach works as well.